cmd/system-probe/modules/network_tracer.go

```diff
@@ -78,13 +80,76 @@ type networkTracer struct {
 	tracer       *tracer.Tracer
 	done         chan struct{}
 	restartTimer *time.Timer
+
+	// Intended to ensure (in compile time) we're implementing the SystemProbeServer interface
+	connectionserver.UnsafeSystemProbeServer
+	maxConnsPerMessage int
 }
 
 func (nt *networkTracer) GetStats() map[string]interface{} {
 	stats, _ := nt.tracer.GetStats()
 	return stats
 }
 
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// GetConnections function that establishes a streaming RPC connection to retrieve and continuously stream information
+// about the current connections in the system.
+func (nt *networkTracer) GetConnections(req *connectionserver.GetConnectionsRequest, s2 connectionserver.SystemProbe_GetConnectionsServer) error {
+	runCounter := atomic.NewUint64(0)
```

Comment:
the run counter is a fake, you're creating it here and log the count later in the function